### PR TITLE
[GCS]Remove unused api(ServiceBasedActorInfoAccessor::AsyncRegister/ServiceBasedActorInfoAccessor::AsyncUpdate)

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
Remove unused api：
ServiceBasedActorInfoAccessor::AsyncRegister
ServiceBasedActorInfoAccessor::AsyncUpdate

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/test/subscription_executor_test.cc [hunk 1]
```diff
@@ -1,225 +0,0 @@
...
-      JobID job_id = JobID::FromInt(i);
-      actor->set_job_id(job_id.Binary());
-      actor->set_state(ActorTableData::ALIVE);
-      ActorID actor_id = ActorID::Of(job_id, RandomTaskId(), /*parent_task_counter=*/i);
-      actor->set_actor_id(actor_id.Binary());
-      id_to_data_[actor_id] = actor;
-    }
-  }
-
-  size_t AsyncRegisterActorToGcs() {
```
Comment (ffbin) [inline src/ray/gcs/test/subscription_executor_test.cc:76]: We deleted `AsyncRegister`, and this test case needs to use x, so we deleted `AsyncRegister`. BTW: SubscriptionExecutor is not used, we will delete it later.

File: src/ray/gcs/gcs_client/test/global_state_accessor_test.cc [hunk 2]
```diff
@@ -229,27 +229,6 @@ TEST_F(GlobalStateAccessorTest, TestObjectTable) {
...
-    actor_ids.emplace_back(ActorID::FromBinary(actor_table_data->actor_id()));
-    std::promise<bool> promise;
-    RAY_CHECK_OK(gcs_client_->Actors().AsyncRegister(
-        actor_table_data, [&promise](Status status) { promise.set_value(status.ok()); }));
-    WaitReady(promise.get_future(), timeout_ms_);
-  }
-  ASSERT_EQ(global_state_->GetAllActorInfo().size(), actor_count);
-
-  for (auto &actor_id : actor_ids) {
-    ASSERT_TRUE(global_state_->GetActorInfo(actor_id));
```
Comment (ffbin) [inline src/ray/gcs/gcs_client/test/global_state_accessor_test.cc:249]: test_global_state.py has test `GetActorInfo` and `GetAllActorInfo`.

**Comment by ffbin** [comment]
Hi @rkooo567 , this PR is ready to be reviewed, thanks.

**Review by rkooo567 — APPROVED** [review]

**Comment by ericl** [comment]
@ffbin looks like this broke the Windows build; please check it before merging.

**Comment by rkooo567** [comment]
Sorry my bad. Thought the test was unrelated. Will be more careful next
time!

**Comment by ffbin** [comment]
> @rkooo567 @ffbin looks like this broke the Windows build; please check it before merging.

Sorry, https://github.com/ray-project/ray/runs/1190596975#step:7:2184 there is no compile failure in the windows build of this PR, so thought the test was unrelated. Will be more careful next time!

**Pull request merged**